tree/dataframe/src/RDFDisplay.cxx

```diff
@@ -232,33 +248,35 @@ void RDisplay::Print() const
                // Thus, the first column is only the Row column and the actual first column is printed
          columnsToPrint = 2;
       }
-      Info("Print", "Only showing %zu columns out of %zu\n", columnsToPrint, fNColumns);
+      ret.fMsgs.push_back(std::string("Only showing ") + std::to_string(columnsToPrint) + " columns out of " +
+                          std::to_string(fNColumns));
```

Comment:
In principle we might want to make `considerDots` an option orthogonal to Print vs AsString. Delegating the decision whether to print the messages or not to the caller allows to change this more easily, and since this code doesn't have strict performance requirements I think we can accept some wasted work when we (currently) ignore the messages in AsString.

Reply:
>  It is also an order of printing change.

You mean that all warnings will now be grouped after the regular printing? That's true, I didn't consider that and I suppose we want to preserve the current printing behavior...I'll adopt your solution, thanks.

Reply:
@silverweed @vepadulano Was this last comment discussed offline?

Reply:
Sorry Philippe, I didn't see your last comment so I thought everything was good. But you're clearly right, the order is not preserved even with your initial solution. The `ToStream` solution sounds good, I will open a PR with that change. Apologies for not addressing this here.

Reply:
Although even with the `ToStream` I would have to either pass 2 streams or filter out the warnings from `AsString()`...so I think the real solution will be slightly more complex.

Reply:
#17173

Reply:
> as the warnings are already all printed before the output 

I am not quite sure how beside maybe the difference in flushing between stderr and stdout, but that's not important.  The important part was to make a decision on the intended behavior and document it.

> Sorry Philippe, I didn't see your last comment

I make it an habit to not merge a PR until all discussion are marked as resolved to avoid missing some of the comments.  

The [Celeritas](https://github.com/celeritas-project/celeritas) team wrote a reasonable formal-ish description on how to handle those resolution.
```
When you've fully implemented the reviewer's comment, you may mark it as resolved without commenting. Do not resolve a conversation if you disagree with the feed: instead, post your view in a follow-on comment and wait for the reviewer to respond. If you comment, whether to supplement your change or to iterate with the reviewer, please do not resolve the conversation since that makes it hard to find your comment.
```

Reply:
> I make it an habit to not merge a PR until all discussion are marked as resolved to avoid missing some of the comments.

I wish Github had a good way to handle reviews, I still haven't found any system that works better than Helix Swarm on this. One can dream.

> I am not quite sure how beside maybe the difference in flushing between stderr and stdout

Simply, all the `Info` calls come before any `cout` call in the original code, they are fully disjoint. And they should stay that way because otherwise the table output would be messed up by intermingled warnings, which is undesirable.

Reply:
> Simply, all the Info calls come before any cout call in the original code, 

That was not obvious from the diff :)

> And they should stay that way because otherwise the table output would be messed up by intermingled warnings, which is undesirable.

This makes sense.  Thanks.